src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md

```diff
@@ -0,0 +1,212 @@
+---
+layout: post
+title: How Mercado Libre optimized for Web Vitals (TBT/FID)
+subhead: Optimizing interactivity of Mercado Libre's product details pages.
+authors:
+  - demianrenzulli
+  - aranhacarlos
+  - joanbaca
+date: 2020-09-01
+hero: hero.jpg
+description: Summary of the work done by Mercado Libre's frontend architecture team to optimize FID, using TBT as a proxy metric.
+tags:
+  - blog
+  - performance
+  - web-vitals
+---
+
+Mercado Libre is the largest e-commerce and payments ecosystem in Latin America. It is present in 18 countries and is a market leader in Brazil, Mexico, and Argentina (based on unique visitors and pageviews).
+
+Web performance has been a focus for the company for a long time. They recently formed a team to monitor performance and apply optimizations across different parts of the site.
+
+This article summarizes the work done by [Guille Paz](https://twitter.com/pazguille), [Pablo Carminatti](https://www.linkedin.com/in/pcarminatti/), and [Oleh Burkhay](https://twitter.com/oburkhay) from Mercado Libre's frontend architecture team to optimize one of the Core Web Vitals: [First Input Delay (FID)](https://web.dev/fid/) and it's lab proxy, [Total Blocking Time (TBT)](https://web.dev/tbt/).
+
+## Long tasks, First Input Delay, and Total Blocking Time
+
+Running expensive Javascript code can lead to [long tasks](https://web.dev/long-tasks-devtools/), which are those that run for more than **50ms** in the browser's main thread.
+
+FID (First Input Delay) measures the time from when a user first interacts with a page (e.g. when they click on a link) to the time when the browser is actually able to begin processing event handlers in response to that interaction. A site that executes expensive Javascript code will likely have several long tasks, which will end up negatively impacting FID.
+
+To provide a good user experience, sites should strive to have a First Input Delay of less than 100 milliseconds:
+
+<picture>
+  <source srcset="../vitals/fid_8x2.svg" media="(min-width: 640px)">
+  <img class="w-screenshot w-screenshot--filled"
+      src="../vitals/fid_4x3.svg"
+      alt="Good fid values are 2.5 seconds, poor values are greater than 4.0
+            seconds and anything in between needs improvement">
+</picture>
+
+While Mercado Libre's site was performing well in most sections, they found in the [Chrome User Experience Report](https://developers.google.com/web/tools/chrome-user-experience-report) that product detail pages had a poor FID. Based on that information, they decided to focus their efforts on improving the interactivity for product pages in the site.
+
+<figure class="w-figure">
+  <img src="meli-pdp.png" 
+       alt="The mobile and desktop versions of a Mercado Libre product detail page.">
+   <figcaption class="w-figcaption">
+      The mobile and desktop versions of a Mercado Libre product detail page.
+  </figcaption>     
+</figure>
```

Comment:
Can you split this up into two images? The [Column compontent](https://web.dev/handbook/web-dev-components/#columns) should help you keep them side-by-side.

And make sure to add `class="w-screenshot"` to each image please

Reply:
Thanks @kaycebasques! I tried the following:

- Breaking the main image into two independent images (one for mobile the other for desktop).
- Using the column component.

In the original version, the mobile screenshot takes around 1/3 of the width. When using the component it assigned half of the width to each image, making the mobile one look much bigger and the desktop one very small.

I played around adding a width attribute to the img element, but it doesn't work either. My options here would be:

- Leaving the image as is.
- Picking only one of the two form factors and cut the other. I think the mobile one would be representative enough.

Let me know what you think.

Reply:
In the original image, mobile occupies 1/3 of the image and desktop 2/3. When breaking into two images, the column component assigns 50% to each, which makes the mobile look much bigger and the desktop much smaller.

I managed to make both look better by adding width="141" to the img element of meli-pdp-mobile.png, but it looks like a hack (perhaps you know a better way of doing this).

In any case, I still like more how it looked like in the original one.

Let me know if you have any ideas for this.

Reply:
The columns you've provided look fine. I mainly wanted to split it up in order to add `class="w-screenshot"` to each